BUGFIX: Incorrect type used for `Swift_Mime_SimpleMessage::setDate()`

mattclegg opened the pull request:
I'm receiving the error;

```Argument 1 passed to Swift_Mime_SimpleMessage::setDate() must implement interface DateTimeInterface, int given, called in /var/www/vendor/silverstripe/framework/src/Control/Email/Email.php on line 272```

I'm also not sure why the Date needs to be set explicitly. Any feedback welcome.

dhensby commented:
> I'm also not sure why the Date needs to be set explicitly. Any feedback welcome.

As you can see from the builds, it's so tests can make assertions about the messages.

Interesting you're getting this error but the test suite doesn't. What version of SwiftMailer are you using?

mattclegg commented:
Thanks for the clarification.

I did have the wrong version installed. composer.json showed that I had installed `v5.4.12` but when I check the source SwiftMailer it was actually v6? I deleted it and updated from composer again and seems to be fixed.
